3-make_gene_matrix.R

```diff
@@ -0,0 +1,122 @@
+
+# Options:
+# "-d" - Directory of where individual samples' salmon folders are located.
+# "-o" - Directory of where the output gene matrix RDS file should go.
+# "-g" - GEO ID for the dataset so the metadata can be downloaded.
+# "-m" - Percent mapped reads (reported as a decimal) cutoff for filtering 
+#        samples. Default is 0.5.
+# "-l" - Optional label to add to output files. Generally necessary if processing
+#        multiple datasets in the same pipeline.
+# 
+# Command line example:
+
+# Rscript scripts/3-make_gene_matrix.R \
+# -d data/salmon_quants \
+# -o data \
+# -g GSE86445 \
+# -m 0.5 \
+# -l "patel"
+
+#-------------------------- Get necessary packages-----------------------------#
+if (!("org.Hs.eg.db" %in% installed.packages())) { 
+  source("https://bioconductor.org/biocLite.R")
+  biocLite("org.Hs.eg.db", suppressUpdates = TRUE)
+}
+if (!("tximport" %in% installed.packages())) {
+  source("https://bioconductor.org/biocLite.R")
+  biocLite("tximport", suppressUpdates = FALSE)
+}
+if (!("rjson" %in% installed.packages())) {
+  install.packages("rjson", suppressUpdates = FALSE)
+}
+
+# Attach needed libraries
+library(org.Hs.eg.db)
+library(optparse)
+library(tximport)
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+#--------------------------------Set up options--------------------------------#
+option_list <- list(
+    make_option(opt_str = c("-d", "--dir"), type = "character", default = getwd(),
+        help = "Directory where salmon quantification folders are located",
+        metavar = "character"),
+    make_option(opt_str = c("-g", "--geo"), type = "character", default = getwd(),
+        help = "GEO ID of dataset for downloading the metadata",
+        metavar = "character"),
+    make_option(opt_str = c("-o", "--output"), type = "character",
+        default = getwd(), help = "Directory where you would like the output to go",
+        metavar = "character"),
+    make_option(opt_str = c("-m", "--mapped"), type = "numeric", 
+        default = "0.5", help = "Cutoff for what percent mapped_reads samples
+        should have. Any samples with less than the cutoff will be removed.",
+        metavar = "numeric"), 
+    make_option(opt_str = c("-l", "--label"), type = "character",
+        default = "", help = "Optional label for output files",
+        metavar = "character")
+)
+
+# Parse options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+#------------------------------Import Salmon reads-----------------------------#
+# Get quant files
+quant.files <- list.files(opt$dir, recursive = TRUE, full.names = TRUE,
+                       pattern = "quant.sf")
+
+# Carry sample names over
+sample.names <- stringr::word(quant.files, 3, sep = "/")
+
+# Get transcript IDs
+transcripts <- read.table(quant.files[[1]], header = TRUE, 
+                 colClasses = c("character", rep("NULL", 4)))
+
+transcripts.wout.ver <- gsub("\\.[0-9]*$", "", transcripts$Name)
```

Comment:
Instead of doing this, I think you can use `ignoreTxVersion = TRUE` in line 94 (double check that, I might have the logic wrong)

Reply:
That's good to know. However, this solution does resolve that `mapIds` needs transcripts without the version number. I looked online to see if they have a similar option but they don't appear to. So unfortunately I think I still have to make this ugly extra `transcripts.wout.ver` Unless you know of a way to get `mapIds` to ignore transcript version?